packages/react-dom/src/__tests__/ReactDOMFloat-test.js

```diff
@@ -28,3718 +34,3862 @@ let hasErrored = false;
       }
-      const root = ReactDOMClient.createRoot(container);
-      root.render(<Component />);
-      expect(Scheduler).toFlushWithoutYielding();
-      expect(getMeaningfulChildren(document)).toEqual(
-        <html>
-          <head>
-            <link rel="preload" as="style" href="foo" />
-          </head>
-          <body>
-            <div id="container">foo</div>
-          </body>
-        </html>,
-      );
-    });
 
-    // @gate enableFloat
-    it('inserts a preload resource when called in a layout effect', async () => {
-      function App() {
-        React.useLayoutEffect(() => {
-          ReactDOM.preload('foo', {as: 'style'});
-        }, []);
-        return 'foobar';
+      function nodeWriteableStreamToWebWritableStream(nodeStream) {
+        const webStream = new WritableStream({
+          write(chunk) {
+            return new Promise((resolve, reject) => {
+              nodeStream.write(chunk, error => {
+                if (error) {
+                  reject(error);
+                } else {
+                  resolve();
+                }
+              });
+            });
+          },
+          close() {
+            return new Promise((resolve, reject) => {
+              nodeStream.end(error => {
+                if (error) {
+                  reject(error);
+                }
+                resolve();
+              });
+            });
+          },
+          abort(error) {
+            return new Promise((resolve, reject) => {
+              nodeStream.destroy(error, () => {
+                if (error) {
+                  reject(error);
+                }
+                resolve();
+              });
+            });
+          },
+        });
+        return webStream;
       }
-      const root = ReactDOMClient.createRoot(container);
-      root.render(<App />);
-      expect(Scheduler).toFlushWithoutYielding();
-
-      expect(getMeaningfulChildren(document)).toEqual(
-        <html>
-          <head>
-            <link rel="preload" as="style" href="foo" />
-          </head>
-          <body>
-            <div id="container">foobar</div>
-          </body>
-        </html>,
-      );
-    });
 
-    // @gate enableFloat
-    it('inserts a preload resource when called in a passive effect', async () => {
-      function App() {
-        React.useEffect(() => {
-          ReactDOM.preload('foo', {as: 'style'});
-        }, []);
-        return 'foobar';
+      function renderToPipeableStream(jsx, options) {
+        if (runtime === 'browser') {
+          const streamPromise = ReactDOMFizzServer.renderToReadableStream(
+            jsx,
+            mergeOptions(options, renderOptions),
+          );
+          return {
+            pipe: destination => {
+              streamPromise.then(stream =>
+                stream.pipeTo(
+                  nodeWriteableStreamToWebWritableStream(destination),
+                ),
+              );
+              return destination;
```

Comment:
It is interesting that this didn't change any test behaviors. For the node runtime you can pipe even before the shell is ready. but in the web stream case we can't start the pipe until the stream promise resolves which happens after the shell is ready. Maybe it doesn't matter too much because nothing flushes until the shell is ready so the early pipe is functionally equivalent. My guess is when we implement preamble this might need to be refactored again.